presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1140 @@
+                conn.namespaceOperations().create(table.getSchema());
+            }
+
+            // Create the table!
+            conn.tableOperations().create(table.getFullTableName());
+
+            // Set locality groups, if any
+            if (groups != null && groups.size() > 0) {
+                Map<String, Set<Text>> localityGroups = new HashMap<>();
+                for (Entry<String, Set<String>> g : groups.entrySet()) {
+                    Set<Text> families = new HashSet<>();
+
+                    for (String col : g.getValue()) {
+                        // Locate the column family from the list of columns
+                        // We already validated this earlier, so it'll exist. Promise.
+                        families.add(new Text(table.getColumns().stream().filter(x -> x.getName().equals(col)).collect(Collectors.toList()).get(0).getFamily()));
+                    }
+                    localityGroups.put(g.getKey(), families);
+                }
+
+                conn.tableOperations().setLocalityGroups(table.getFullTableName(), localityGroups);
+                LOG.info("Set locality groups to %s", groups);
+            }
+            else {
+                LOG.info("No locality groups set");
+            }
+
+            // Create index tables, if appropriate
+            createIndexTables(table);
+            return table;
+        }
+        catch (Exception e) {
+            dropTable(table);
+            throw new PrestoException(INTERNAL_ERROR,
+                    "Accumulo error when creating table, check coordinator logs for more detail",
+                    e);
+        }
+    }
+
+    /**
+     * Auto-generates the mapping of Presto column name to Accumulo family/qualifier, respecting the locality groups (if any).
+     *
+     * @param columns Presto columns for the table
+     * @param groups Mapping of locality groups to a set of Presto columns, or null if none
+     * @return Column mappings
+     */
+    private Map<String, Pair<String, String>> autoGenerateMapping(List<ColumnMetadata> columns, Map<String, Set<String>> groups)
+    {
+        Map<String, Pair<String, String>> mapping = new HashMap<>();
+        for (ColumnMetadata column : columns) {
+            boolean tryAgain = false;
+            do {
+                String fam = null;
+                String qual = UUID.randomUUID().toString().substring(0, 4);
+
+                // search through locality groups to find if this column has a locality group
+                if (groups != null) {
+                    for (Entry<String, Set<String>> g : groups.entrySet()) {
+                        if (g.getValue().contains(column.getName())) {
+                            fam = g.getKey();
+                        }
+                    }
+                }
+
+                // randomly generate column family if not found
+                if (fam == null) {
+                    fam = UUID.randomUUID().toString().substring(0, 4);
```

Comment:
Extracted this to a function.  Using random number generator instead of UUID generation.